ambassador/ambassador/config.py

```diff
@@ -1107,12 +1108,16 @@ def generate_intermediate_config(self):
         if tmod:
             # self.logger.debug("USING TLS")
             primary_listener['tls'] = tmod
+            if self.tmod_certs_exist(primary_listener['tls']) > 0:
+                primary_listener['tls']['ssl_context'] = True
             redirect_cleartext_from = tmod.get('redirect_cleartext_from')
 
         self.envoy_config['listeners'] = [ primary_listener ]
 
         if redirect_cleartext_from:
-            primary_listener['require_tls'] = True
+            # We only want to set `require_tls` on the primary listener when certs are present on the pod
+            if self.tmod_certs_exist(primary_listener['tls']) > 0:
+                primary_listener['require_tls'] = True
```

Comment:
@kflynn does this look reasonable?
When we talk about L4 redirection, we want the HTTP requests that arrive at 443 to go right through, for that we don't need `require_tls`.
Does some use-case come to your mind that could break with this change?

Reply:
I think this is OK, yeah. Had to think about it for a bit.